File: app/src/main/java/com/example/ui/components/CategoryFilterTabs.kt
package com.example.ui.components

import androidx.compose.foundation.horizontalScroll
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.BakeryDining
import androidx.compose.material.icons.filled.Coffee
import androidx.compose.material.icons.filled.GridView
import androidx.compose.material.icons.filled.LocalBar
import androidx.compose.material.icons.filled.LunchDining
import androidx.compose.material.icons.filled.Restaurant
import androidx.compose.material.icons.filled.SetMeal
import androidx.compose.material3.FilterChip
import androidx.compose.material3.FilterChipDefaults
import androidx.compose.material3.Icon
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.platform.testTag
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.example.data.model.CategoryEntity
import com.example.ui.theme.PurplePrimary
import com.example.ui.theme.SurfaceLightChip
import com.example.ui.theme.TextSecondaryLight

@Composable
fun CategoryFilterTabs(
    categories: List<CategoryEntity>,
    selectedCategoryId: Int?,
    onCategorySelected: (Int?) -> Unit,
    modifier: Modifier = Modifier
) {
    Row(
        modifier = modifier
            .fillMaxWidth()
            .horizontalScroll(rememberScrollState())
            .padding(horizontal = 16.dp, vertical = 6.dp),
        horizontalArrangement = Arrangement.spacedBy(8.dp)
    ) {
        // "Semua Menu" chip
        FilterChip(
            selected = selectedCategoryId == null,
            onClick = { onCategorySelected(null) },
            label = {
                Text(
                    text = "Semua Menu",
                    fontSize = 13.sp,
                    fontWeight = if (selectedCategoryId == null) FontWeight.Bold else FontWeight.Medium
                )
            },
            leadingIcon = {
                Icon(
                    imageVector = Icons.Default.GridView,
                    contentDescription = null,
                    modifier = Modifier.size(16.dp)
                )
            },
            shape = RoundedCornerShape(24.dp),
            border = null,
            colors = FilterChipDefaults.filterChipColors(
                selectedContainerColor = PurplePrimary,
                selectedLabelColor = Color.White,
                selectedLeadingIconColor = Color.White,
                containerColor = SurfaceLightChip,
                labelColor = TextSecondaryLight,
                iconColor = TextSecondaryLight
            ),
            modifier = Modifier.testTag("category_chip_all")
        )

        categories.forEach { category ->
            val isSelected = selectedCategoryId == category.id
            FilterChip(
                selected = isSelected,
                onClick = { onCategorySelected(category.id) },
                label = {
                    Text(
                        text = category.name,
                        fontSize = 13.sp,
                        fontWeight = if (isSelected) FontWeight.Bold else FontWeight.Medium
                    )
                },
                leadingIcon = {
                    Icon(
                        imageVector = getCategoryIcon(category.iconName),
                        contentDescription = null,
                        modifier = Modifier.size(16.dp)
                    )
                },
                shape = RoundedCornerShape(24.dp),
                border = null,
                colors = FilterChipDefaults.filterChipColors(
                    selectedContainerColor = PurplePrimary,
                    selectedLabelColor = Color.White,
                    selectedLeadingIconColor = Color.White,
                    containerColor = SurfaceLightChip,
                    labelColor = TextSecondaryLight,
                    iconColor = TextSecondaryLight
                ),
                modifier = Modifier.testTag("category_chip_${category.id}")
            )
        }
    }
}

private fun getCategoryIcon(iconName: String): ImageVector {
    return when (iconName) {
        "Restaurant" -> Icons.Default.Restaurant
        "LunchDining" -> Icons.Default.LunchDining
        "SetMeal" -> Icons.Default.SetMeal
        "LocalBar" -> Icons.Default.LocalBar
        "Coffee" -> Icons.Default.Coffee
        "BakeryDining" -> Icons.Default.BakeryDining
        else -> Icons.Default.Restaurant
    }
}

File: lib/widgets/category_filter_tabs.dart
import 'package:flutter/material.dart';
import '../models/category_model.dart';
import '../theme/app_colors.dart';

class CategoryFilterTabs extends StatelessWidget {
  final List<CategoryModel> categories;
  final int? selectedCategoryId;
  final ValueChanged<int?> onCategorySelected;

  const CategoryFilterTabs({
    super.key,
    required this.categories,
    required this.selectedCategoryId,
    required this.onCategorySelected,
  });

  IconData _getCategoryIcon(String iconName) {
    switch (iconName) {
      case 'Restaurant':
        return Icons.restaurant;
      case 'LunchDining':
        return Icons.lunch_dining;
      case 'SetMeal':
        return Icons.set_meal;
      case 'LocalBar':
        return Icons.local_bar;
      case 'Coffee':
        return Icons.coffee;
      case 'BakeryDining':
        return Icons.bakery_dining;
      default:
        return Icons.restaurant;
    }
  }

  @override
  Widget build(BuildContext context) {
    return SingleChildScrollView(
      scrollDirection: Axis.horizontal,
      padding: const EdgeInsets.symmetric(horizontal: 16, vertical: 6),
      child: Row(
        children: [
          // "Semua Menu" chip
          Padding(
            padding: const EdgeInsets.only(right: 8),
            child: FilterChip(
              selected: selectedCategoryId == null,
              onSelected: (_) => onCategorySelected(null),
              avatar: Icon(
                Icons.grid_view,
                size: 16,
                color: selectedCategoryId == null ? Colors.white : AppColors.textSecondaryLight,
              ),
              label: Text(
                'Semua Menu',
                style: TextStyle(
                  fontSize: 13,
                  fontWeight: selectedCategoryId == null ? FontWeight.bold : FontWeight.w500,
                  color: selectedCategoryId == null ? Colors.white : AppColors.textSecondaryLight,
                ),
              ),
              backgroundColor: AppColors.surfaceLightChip,
              selectedColor: AppColors.primary,
              showCheckmark: false,
              shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(24)),
              side: BorderSide.none,
            ),
          ),

          // Dynamic Category Chips
          ...categories.map((category) {
            final isSelected = selectedCategoryId == category.id;
            return Padding(
              padding: const EdgeInsets.only(right: 8),
              child: FilterChip(
                selected: isSelected,
                onSelected: (_) => onCategorySelected(category.id),
                avatar: Icon(
                  _getCategoryIcon(category.iconName),
                  size: 16,
                  color: isSelected ? Colors.white : AppColors.textSecondaryLight,
                ),
                label: Text(
                  category.name,
                  style: TextStyle(
                    fontSize: 13,
                    fontWeight: isSelected ? FontWeight.bold : FontWeight.w500,
                    color: isSelected ? Colors.white : AppColors.textSecondaryLight,
                  ),
                ),
                backgroundColor: AppColors.surfaceLightChip,
                selectedColor: AppColors.primary,
                showCheckmark: false,
                shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(24)),
                side: BorderSide.none,
              ),
            );
          }),
        ],
      ),
    );
  }
}
